library/core/src/mem/maybe_uninit.rs

```diff
@@ -1200,21 +1333,20 @@ impl<T> MaybeUninit<T> {
     ///
     /// # Examples
     ///
-    /// Fill an uninit vec with the default value.
     /// ```
     /// #![feature(maybe_uninit_fill)]
     /// use std::mem::MaybeUninit;
     ///
-    /// let mut buf = vec![MaybeUninit::<i32>::uninit(); 10];
-    /// let initialized = MaybeUninit::fill_with(buf.as_mut_slice(), Default::default);
+    /// let mut buf = [const { MaybeUninit::<i32>::uninit() }; 10];
+    /// let initialized = buf.write_with(Default::default);
     /// assert_eq!(initialized, &mut [0; 10]);
     /// ```
     #[unstable(feature = "maybe_uninit_fill", issue = "117428")]
-    pub fn fill_with<'a, F>(this: &'a mut [MaybeUninit<T>], mut f: F) -> &'a mut [T]
+    pub fn write_with<F>(&mut self, mut f: F) -> &mut [T]
```

Comment:
I initially did `write_filled_with`, but figured that fewer words would be desirable here since these method names are already quite long. But, since my main goal here is to push these methods in some form and they can be renamed before stabilisation, I'm fine renaming them to whatever people prefer.

Reply:
SGTM, I'll add this as an unresolved question to the tracking issue.